src/index.ts

```diff
@@ -115,7 +118,7 @@ export class StateSource<T> {
   public select<R>(scope: Scope<T, R>): StateSource<R> {
     const get = makeGetter(scope);
     return new StateSource<R>(
-      this.state$.map(get).filter(s => typeof s !== 'undefined'),
+      xs.fromObservable<T>(this.state$).map(get).filter(s => typeof s !== 'undefined'),
```

Comment:
We can avoid this one since RxJS/xstream/most.js all have `map` and `filter`. It should save us something in performance.